taskcat/stacker.py

```diff
@@ -986,12 +1007,19 @@ def generate_input_param_values(self, s_parms, region):
                         _parameters['ParameterValue'] = param_value
 
                 if genaz_single_re.search(param_value):
-                    print(PrintMsg.DEBUG + "Selecting availability zones")
-                    print(PrintMsg.DEBUG + "Requested 1 az")
-                    param_value = self.get_available_azs(
-                        region,
-                        1)
-                    _parameters['ParameterValue'] = param_value
+                    az_id = int(
+                        self.regxfind(count_re, param_value))
+                    if az_id:
+                        print(PrintMsg.DEBUG + "Requested 1 az")
+                        print(PrintMsg.DEBUG + "Selecting availability zone %s" % az_id)
+                        param_value = self.get_single_az(region, az_id)
+                        _parameters['ParameterValue'] = param_value
+                    else:
+                        print(PrintMsg.INFO + "$[taskcat_getsingleaz_(!)]")
+                        print(PrintMsg.INFO + "Which az not specified!")
+                        print(PrintMsg.INFO + " - (Defaulting to az 1)")
+                        param_value = self.get_single_az(region, 1)
```

Comment:
might be nice to return a random az in this case ?

Reply:
No longer applicable, per our discussion below and the latest push.